src/modules/network/components/DialogAddNetwork/DialogAddNetwork.js

```diff
@@ -1,83 +1,109 @@
-import React, { useState } from 'react';
+import React, { useMemo, useState } from 'react';
 import { useHistory } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+import { useForm } from 'react-hook-form';
+import { toast } from 'react-toastify';
 import Dialog from '@theme/dialog/dialog';
 import Box from '@theme/box';
 import BoxHeader from '@theme/box/header';
 import BoxContent from '@theme/box/content';
-import classNames from 'classnames';
-import { useForm } from 'react-hook-form';
 import { parseSearchParams } from 'src/utils/searchParams';
 import Input from '@theme/Input';
-import { PrimaryButton } from '@theme/buttons';
+import { PrimaryButton, TertiaryButton } from '@theme/buttons';
 import useSettings from '@settings/hooks/useSettings';
-import { immutablePush, immutableSetToArray } from 'src/utils/immutableUtils';
+import { immutableSetToArray } from 'src/utils/immutableUtils';
 import { regex } from 'src/const/regex';
+import {
+  DEFAULT_NETWORK_FORM_STATE,
+  getDuplicateNetworkFields,
+} from '@network/components/DialogAddNetwork/utils';
+import { useNetworkStatus } from '@network/hooks/queries';
+import { Client } from 'src/utils/api/client';
 import networks from '../../configuration/networks';
 import styles from './DialogAddNetwork.css';
 
+// eslint-disable-next-line max-statements,complexity
 const DialogAddNetwork = () => {
+  const { t } = useTranslation();
   const history = useHistory();
-  const { name: defaultName = '', serviceUrl: defaultServiceUrl = '' } = parseSearchParams(
-    history.location.search
-  );
   const { setValue, customNetworks } = useSettings('customNetworks');
-  const [successText, setSuccessText] = useState('');
+  const [isAddingNetwork, setIsAddingNetwork] = useState(false);
+  const [isNetworkUrlOk, setIsNetworkUrlOk] = useState(true);
   const [errorText, setErrorText] = useState('');
 
-  const { t } = useTranslation();
+  const { name: defaultName = '' } = parseSearchParams(history.location.search);
+
   const {
+    watch,
     register,
     handleSubmit,
     reset,
-    formState: { errors },
+    formState: { errors, isDirty },
   } = useForm({
-    defaultValues: {
-      name: defaultName,
-      serviceUrl: defaultServiceUrl,
-    },
+    defaultValues: defaultName
+      ? customNetworks.find((customNetwork) => customNetwork.name === defaultName)
+      : DEFAULT_NETWORK_FORM_STATE,
+  });
+  const formValues = watch();
+
+  const client = useMemo(
+    () => formValues.serviceUrl && new Client({ http: formValues.serviceUrl }),
+    [formValues.serviceUrl]
+  );
+
+  const networkStatus = useNetworkStatus({
```

Comment:
I made a mistake with networkStatus hook suggestion.. here is the problem.

- One of the most important endpoints wallets relay on is off-chain metadata endpoint, if it doesnt respond the wallet wont function
- Ex. use http://betanet-service.liskscan.com/ and add to network and success try to switch
Since the metadata endpoint is failing the wallet wont work.

So i suggest the following, we check both onchain and offchain.. so use the blockchain apps meta to know if the network is actually working while adding.. it can happen that it might fail later, but atleast we can ensure its working while adding/editing?

```
  const networkStatus = useNetworkStatus({
    options: { enabled: !!mainChainNetwork },
    client: queryClient.current,
  });

  const blockchainAppsMeta = useBlockchainApplicationMeta({
    config: {
      params: {
        chainID: [...new Set([networkStatus.data?.data?.chainID, currentApplication.chainID])]
          .filter((item) => item)
          .join(','),
      },
    },
    options: { enabled: !!networkStatus.data && !!mainChainNetwork },
    client: queryClient.current,
  });
```

here is the reference https://github.com/LiskHQ/lisk-desktop/blob/a9ee91f76730c26b245745683be8436fc665fab4/setup/react/app/ApplicationBootstrap.js

Reply:
this not yet yet done